src/libmongoc/src/mongoc/mongoc-client-side-encryption.h

```diff
@@ -33,10 +33,16 @@ struct _mongoc_cursor_t;
 #define MONGOC_ENCRYPT_ALGORITHM_INDEXED "Indexed"
 #define MONGOC_ENCRYPT_ALGORITHM_UNINDEXED "Unindexed"
 
+#define MONGOC_ENCRYPT_QUERY_TYPE_EQUALITY "equality"
+
+
 BSON_BEGIN_DECLS
 
 typedef struct _mongoc_auto_encryption_opts_t mongoc_auto_encryption_opts_t;
 
+typedef bool (*mongoc_kms_credentials_provider_callback_fn) (
+   void *userdata, bson_t *out, bson_error_t *error);
```

Comment:
```suggestion
   void *userdata, bson_t *out, void* reserved, bson_error_t *error);
```

Consider adding an argument for future extension. For example: libmongocrypt does not distinguish which KMS provider credentials are being requested. That was discussed and [determined out of scope](https://github.com/kevinAlbs/mongo-go-driver/pull/1#issuecomment-1058214761). If that is added in the future, it may be convenient to pass another handle to the callback.

Reply:
Should I give it a stronger type than `void*`? i.e. `const bson_t* params`

Reply:
`const bson_t* params` SGTM.

I thought it may not be ABI / API breaking if a `void*` argument is later changed to a non-void pointer. [Further experimentation](https://godbolt.org/z/dWW3j7a6x) suggest that is incorrect.